### ISSUE TITLE
Model.query/query_delete fails with large amounts of existing Models

### ISSUE THREAD

**Issue by Frank5000** [coleifer/walrus#118]
TLDR: Running query_delete() with tens of thousands of hits expected only ends up getting ~6.5k

The setup is all in docker-compose, with multiple container processes creating and deleting Models in a shared Redis instance (5.0.6). Each process only deletes its own models to avoid the KeyError thrown when Walrus is unable to do a `load(pk)`.

Each model `A` has a field `origin` identifying which container process it was created from and a DateTimeField `created_timestamp` denoting the creation timestamp:
```
class BaseModel(Model):
    __database__ = db
    __namespace__ = 'asd'
    pk = UUIDField(primary_key=True)
    created_timestamp = DateTimeField(default=datetime.datetime.now, index=True)

class A(BaseModel):
    origin = TextField(index=True)
```

The idea is each container prunes models created by it before a certain threshold (600s ago) like so:

```
threshold = datetime.datetime.now() - datetime.timedelta(seconds=600)
A.query_delete((A.origin == "current container") & (A.created_timestamp <= threshold))
```

This should delete ~75k model instances but each run only gets a small percentage (~7k). I've viewed the data in Redis to verify that the timestamps etc are valid. 

Strangely enough, this issue was resolved by sticking the query_delete in a loop until no expired instances remain:

```
while True:
    A.query_delete((A.origin == "current container") & (A.created_timestamp <= threshold))
    
    remaining = sum([1 for _ in A.query((A.origin == "current container") & (A.created_timestamp <= threshold))
    if remaining == 0:
        break
```

Obviously, this solution suboptimal and any insight would be appreciated. Could this be a limitation with Redis? 

Big fan of this project and appreciate the good work!

**Comment by coleifer** [comment]
Walrus may store the results of filter expressions to speed-up operations. See the `_zset_score_filter` method in the `query` module.

In short, walrus models make a tradeoff that seems to be causing you to get cached results. I'd strongly suggest using a different tool or library for this kind of functionality ([see note](https://walrus.readthedocs.io/en/latest/models.html#need-more-power)).

Apologies for the inconvenience. Although the models APIs were conceived with good intentions, in practice I have found them to be a very poor fit for Redis. It's far better *and* more efficient to use the builtin data-types directly, as the walrus models spew tons of data into secondary indices and the-like, attempting to provide a generic solution to an already-solved problem (see: sql). Play to redis' strengths and use the data-types directly. A zset with a score for the timestamp should suffice - you can probably point those zset keys to a hash w/your data if you need to. Or just use sql.

**Comment by Frank5000** [comment]
Yeah it's a shame python-stdnet seems to be dead. Thanks for the response!
